src/emulator/gui/src/game_selector.cpp

```diff
@@ -179,22 +190,105 @@ void draw_game_selector(GuiState &gui, HostState &host) {
+                                if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
+                                    gui.current_background = gui.user_backgrounds[host.cfg.background_image];
+                                else if (!gui.user_backgrounds[host.cfg.background_image] && gui.current_background != 0)
+                                    gui.current_background = 0;
+                            } else
+                                LOG_ERROR("Deleting the game '{} [{}]' have failed.", game.title_id.c_str(), game.title.c_str());
+                            gui.delete_game_popup = false;
+                            ImGui::CloseCurrentPopup();
+                        }
+                        ImGui::SameLine();
+                        if (ImGui::Button("No", CONTEXT_BUTTON_SIZE)) {
+                            gui.delete_game_popup = false;
+                            ImGui::CloseCurrentPopup();
+                        }
+                        ImGui::EndPopup();
+                    } else
+                        ImGui::PopStyleColor();
                 }
+
             }
             ImGui::NextColumn();
-            ImGui::Selectable(game.title_id.c_str(), &selected[1], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, icon_size));
+            ImGui::Selectable(game.title_id.c_str(), &selected[1], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, static_cast<float>(icon_size)));
+            if (ImGui::IsItemHovered() && gui.title_id != game.title_id)
+                gui.title_id = game.title_id;
+            ImGui::OpenPopupOnItemClick("#game_context_menu", 1);
             ImGui::NextColumn();
-            ImGui::Selectable(game.app_ver.c_str(), &selected[2], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, icon_size));
+            ImGui::Selectable(game.app_ver.c_str(), &selected[2], ImGuiSelectableFlags_SpanAllColumns, ImVec2(0, static_cast<float>(icon_size)));
             ImGui::NextColumn();
             if (ImGui::IsItemHovered()) {
-                if (host.cfg.show_game_background) {
-                    if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
-                        gui.current_background = gui.user_backgrounds[host.cfg.background_image];
-                    else if (!gui.user_backgrounds[host.cfg.background_image] && gui.current_background != 0)
-                        gui.current_background = 0;
-                }
+                if (gui.user_backgrounds[host.cfg.background_image] && gui.current_background != static_cast<std::uint32_t>(gui.user_backgrounds[host.cfg.background_image]))
+                    gui.current_background = gui.user_backgrounds[host.cfg.background_image];
+                else if (!gui.user_backgrounds[host.cfg.background_image] && gui.current_background != 0)
```

Comment:
No checking if the value is `0` here.

Reply:
No is needed same other, i not want stupid things like spam set background, it est only one time and that all , est un on old pr exactly for fix this and is so good

Reply:
Zang... I'm very sad about this.

Did you know that setting the background probably takes so many **nanoseconds** on your computer. It does not take that long!

It is probably 100x faster than the ImGui calls you are also calling 60 times a second. You are not saving performance here. It's just redundant :P

Reply:
code is arleady changed, but sorry, i no remove this check for tjat, i have set this on one commit for fix probleme becayse before it spaming re set value